src/renwindow.c

```diff
@@ -24,64 +24,55 @@ static void setup_renderer(RenWindow *ren, int w, int h) {
     SDL_DestroyTexture(ren->texture);
   }
   ren->texture = SDL_CreateTexture(ren->renderer, SDL_PIXELFORMAT_BGRA32, SDL_TEXTUREACCESS_STREAMING, w, h);
-  ren->surface_scale = query_surface_scale(ren);
+  ren->rs.scale = query_surface_scale(ren);
 }
 #endif
 
 
 void renwin_init_surface(UNUSED RenWindow *ren) {
 #ifdef LITE_USE_SDL_RENDERER
-  if (ren->surface) {
-    SDL_FreeSurface(ren->surface);
+  if (ren->rs.surface) {
+    SDL_FreeSurface(ren->rs.surface);
   }
   int w, h;
   SDL_GL_GetDrawableSize(ren->window, &w, &h);
-  ren->surface = SDL_CreateRGBSurfaceWithFormat(0, w, h, 32, SDL_PIXELFORMAT_BGRA32);
-  if (!ren->surface) {
+  ren->rs.surface = SDL_CreateRGBSurfaceWithFormat(0, w, h, 32, SDL_PIXELFORMAT_BGRA32);
+  if (!ren->rs.surface) {
     fprintf(stderr, "Error creating surface: %s", SDL_GetError());
     exit(1);
   }
   setup_renderer(ren, w, h);
 #endif
 }
 
-int renwin_surface_scale(UNUSED RenWindow *ren) {
-#ifdef LITE_USE_SDL_RENDERER
-  return ren->surface_scale;
-#else
-  return 1;
-#endif
-}
-
 
 static RenRect scaled_rect(const RenRect rect, const int scale) {
   return (RenRect) {rect.x * scale, rect.y * scale, rect.width * scale, rect.height * scale};
 }
 
 
 void renwin_clip_to_surface(RenWindow *ren) {
-  SDL_Surface *surface = renwin_get_surface(ren);
-  SDL_SetClipRect(surface, NULL);
+  SDL_SetClipRect(renwin_get_surface(ren).surface, NULL);
 }
 
 
 void renwin_set_clip_rect(RenWindow *ren, RenRect rect) {
-  SDL_Surface *surface = renwin_get_surface(ren);
-  RenRect sr = scaled_rect(rect, renwin_surface_scale(ren));
-  SDL_SetClipRect(surface, &(SDL_Rect){.x = sr.x, .y = sr.y, .w = sr.width, .h = sr.height});
+  RenSurface rs = renwin_get_surface(ren);
+  RenRect sr = scaled_rect(rect, rs.scale);
+  SDL_SetClipRect(rs.surface, &(SDL_Rect){.x = sr.x, .y = sr.y, .w = sr.width, .h = sr.height});
 }
 
 
-SDL_Surface *renwin_get_surface(RenWindow *ren) {
+RenSurface renwin_get_surface(RenWindow *ren) {
 #ifdef LITE_USE_SDL_RENDERER
-  return ren->surface;
+  return ren->rs;
 #else
   SDL_Surface *surface = SDL_GetWindowSurface(ren->window);
   if (!surface) {
     fprintf(stderr, "Error getting window surface: %s", SDL_GetError());
     exit(1);
   }
-  return surface;
+  return (RenSurface){.surface = surface, .scale = 1};
```

Comment:
Can we use a proper variable for this, instead of casting an anonymous struct?

Reply:
Testing has shown that anonymous structs being casted will force every uninitialized variable to 0.

Since there are no uninitialied variables here, there is no downside to doing this.